ZombieSharp/Plugin/Events/EventsExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿namespace ZombieSharp.Plugin.Events;
+
+using CounterStrikeSharp.API.Core;
+
+public static class EventsExtensions
+{
+    private static readonly List<IEventBase> _events = [
```

Comment:
Need a source gen

Reply:
what source gen?

Reply:
The thing that allows to write code which generator another code. Here I want to implement custom attribute `[Listener]` or `[Event]` and mark class with it:
```cs
[Event]
public class PlayerHurtEvent : EventBase<EventPlayerHurt>
{
    public override GameEventHandler<EventPlayerHurt> Handler => OnPlayerHurt;

    private HookResult OnPlayerHurt(EventPlayerHurt @event, GameEventInfo info)
    {
        Console.WriteLine($"Player {@event.Userid} hurt with {@event.DmgHealth} damage.");
        return HookResult.Continue;
    }
}
```
then it should analyze code and take each class with attribute `[Event]`. Then we have a collection of these classes and we can use them with existing:
```diff
public static class EventsExtensions
{
-   private static readonly List<IEventBase> _events = [
-      new PlayerHurtEvent(),
-      // other events
-   ];
+   private static readonly List<IEventBase> _events = SourceGenerators.Events; // collection of events provided by source gen

    public static void RegisterAllEventHandlers(this BasePlugin plugin)
    {
        foreach (var @event in _events)
        {
            @event.Register(plugin);
        }
    }

    public static void RemoveAllEventHandlers(this BasePlugin plugin)
    {
        foreach (var eventInstance in _events)
        {
            eventInstance.Deregister(plugin);
        }
    }
}

Reply:
The output of source gen should be
```cs
public static partial class SourceGenerators {
    public static readonly List<IEventBase> = [
        // all events here which were created by source generator, but not manual resolving
    ];
}
```